app/src/components/TransferItems.js

```diff
@@ -0,0 +1,364 @@
+    }
+    case 'UPDATE': {
+      const { transferItem, updatedTransferItem } = payload
+      const index = state.indexOf(transferItem)
+
+      return [
+        ...state.slice(0, index),
+        updatedTransferItem,
+        ...state.slice(index + 1),
+      ]
+    }
+    case 'REMOVE': {
+      const { transferItem } = payload
+      if (state.length === 1) {
+        return initTransferItems()
+      } else {
+        return state.filter(t => t !== transferItem)
+      }
+    }
+    case 'REMOVE_ALL': {
+      return initTransferItems()
+    }
+    default: {
+      return state
+    }
+  }
+}
+
+const TransferItems = React.memo(
+  React.forwardRef(({ tokens, onSubmit }, ref) => {
+    const { api } = useAragonApi()
+    const accountsRef = useRef()
+    const fieldsLayout = useFieldsLayout(tokens)
+
+    const [transferItems, setTransferItems] = useReducer(
+      transferItemsReducer,
+      null,
+      initTransferItems
+    )
+    const errors = validateFormItems(transferItems)
+    const [addressErrors, setAddressErrors] = useState([])
+    const showDeleteAll = transferItems.length > 1
+
+    const [focusLastAccountNext, setFocusLastAccountNext] = useState(false)
+
+    const addAccount = () => {
+      setTransferItems({
+        type: 'ADD',
+      })
+      setTimeout(() => {
+        focusLastAccount()
+      }, 0)
+    }
+
+    const removeAccount = transferItem => () => {
+      setTransferItems({
+        type: 'REMOVE',
+        payload: {
+          transferItem,
+        },
+      })
+    }
+
+    const removeAllAccounts = () => {
+      setTransferItems({
+        type: 'REMOVE_ALL',
+      })
+      setTimeout(() => {
+        focusLastAccount()
+      }, 0)
+    }
+
+    const updateAccount = transferItem => updatedTransferItem => {
+      setTransferItems({
+        type: 'UPDATE',
+        payload: {
+          transferItem,
+          updatedTransferItem,
+        },
+      })
+    }
+
+    const handleSubmit = async () => {
+      if (errors.length !== 0) return
+
+      const _transferItems = await Promise.all(
+        transferItems.map(async transferItem => ({
+          ...transferItem,
+          address: await searchIdentity(api, transferItem.account),
+        }))
+      )
+      const addressErrors = validateAddresses(
+        _transferItems.map(({ address }) => address)
+      )
+
+      if (addressErrors.length > 0) {
+        setAddressErrors(addressErrors)
+      } else {
+        setAddressErrors([])
+        await onSubmit(_transferItems)
+        setTransferItems({
+          type: 'REMOVE_ALL',
+        })
+      }
+    }
+
+    const handlePaste = transferItem => pasteData => {
+      try {
+        let parsedItems = csvStringToArray(pasteData)
+        if (parsedItems[0][1] === undefined) {
+          parsedItems = csvStringToArray(pasteData, ',')
+        }
+        if (parsedItems.length === 1 && parsedItems[0].length === 1)
+          throw new Error('just string')
```

Comment:
This and the `Error` in line 177 are the errors I think should be more descriptive.